zerver/views/user_groups.py

```diff
@@ -26,6 +26,11 @@ def add_user_group(request: HttpRequest, user_profile: UserProfile,
     check_add_user_group(user_profile.realm, name, user_profiles, description)
     return json_success()
 
+def error_msg(user_profile: UserProfile, user_group: UserGroup, msg: Text) -> Any:
+    group_member_ids = get_user_group_members(user_group)
+    if (not user_profile.is_realm_admin and user_profile.id not in group_member_ids):
+        return JsonableError(msg)
```

Comment:
This should just raise the JsonableError rather than returning; them you don't need to have the callers do `if error is not None`.

I'd also move this into `access_user_group_by_id` (and passing in a `UserProfile`) rather than having a separate 
function.

Finally, I don't think we need separate error messages for each case.  It's a waste of translators' time.  We should just have all 4 of these cases be

"Only group members and organization administrators can administer this user group."

Reply:
possible amendment: I think it might be reasonable to use "group" instead of "user group" when the context is clear. So "Only group members and organization administrators can administer this group."?

Reply:
@rishig I'll do that.